2019-spring/01. Model fitting (NLLS)/nlls.tex

```diff
@@ -0,0 +1,793 @@
+  Мы можем воспользоваться тем, что $\mathbf{H}_{pp}$ легко обратить\footnote[frame]{На самом деле легко решать системы вида $\mathbf{H}_{pp}\mathbf{x}=\mathbf{b}$}.
+  $$
+  \begin{cases}
+    \mathbf{H}_{cc}\mathbf{x}_c+\mathbf{H}_{cp}\mathbf{x}_p=\mathbf{b}_c \\
+    \mathbf{H}_{cp}^\intercal\mathbf{x}_c+\mathbf{H}_{pp}\mathbf{x}_p=\mathbf{b}_b
+  \end{cases}
+  $$
+  \begin{equation*}
+    \begin{gathered}
+    \mathbf{x}_{p}=\mathbf{H}_{pp}^{-1}\mathbf{b}_p-\mathbf{H}_{pp}^{-1}\mathbf{H}_{cp}^\intercal\mathbf{x}_c \\
+       \left(\mathbf{H}_{cc}-\mathbf{H}_{cp}\mathbf{H}_{pp}^{-1}\mathbf{H}_{cp}^\intercal\right)\mathbf{x}_c=\mathbf{b}_c-\mathbf{H}_{cp}\mathbf{H}_{pp}^{-1}\mathbf{b}_p \\
+    \end{gathered}
+  \end{equation*}
+\end{frame}
+
+% \begin{frame}
+  % \frametitle{Расчёт погрешностей}
+  % \framesubtitle{Распространение неопределенности}
+  % $\mathbf{X}\sim\mathcal{N}\left(\overline{\mathbf{X}}, \Sigma\right) \implies \mathbf{A}\mathbf{X}+\mathbf{b}\sim\mathcal{N}\left(\mathbf{A}\overline{\mathbf{X}}+\mathbf{b},\mathbf{A}^\intercal\Sigma\mathbf{A}\right)$
+
+  % Вспомним, что мы приближали нашу функцию остатков как $\mathbf{r}\left(\mathbf{x}+\Delta\mathbf{x}\right)\approx\mathbf{r}\left(\mathbf{x}\right)+\mathbf{J}\Delta\mathbf{x}$
+
+  % Если мы знаем ошибки в измеряемых величинах и учитыаем их ($\mathrm{Cov}\left[\mathbf{r},\mathbf{r}\right]=\mathbf{I}$), то мы можем оценить ошибки в искомых параметрах:
+  % $$\mathrm{Cov}\left[\Delta\mathbf{x}_{last},\Delta\mathbf{x}_{last}\right]=\left(\mathbf{J}^\intercal\mathbf{J}\right)^{-1}$$
+% \end{frame}
+
+\begin{frame}
+  \frametitle{Оценки наибольшего правдоподобия}
+  \framesubtitle{Выбор весов остатков}
+  Пусть мы проводим $n$ измерений неизвестной величины $x\in\mathbb{R}$ с ошибками $\varepsilon_i\sim\mathcal{N}\left(0,\sigma_i^2\right)$ и получаем значения $x_i=x+\varepsilon_i$.
+  
+  Если бы мы знали $x$, плотность вероятности получения именно $\left\{x_i\right\}$ вычислялась бы как
+  $$p\left(\left.\left\{x_i\right\}\right\rvert x\right)=\prod_{i=1}^n\frac{1}{2\sigma_i\sqrt{2\pi}}\exp\left\{-\frac{\left(x-x_i\right)^2}{2\sigma_i^2}\right\}$$
+
+  Будем искать $\overline{x}$, который её максимизирует
+
+  \begin{align*}
+    \overline{x} & =\argmax_{x}p\left(\left.\left\{x_i\right\}\right\rvert x\right) \\
+                 & =\argmin_{x}\sum_{i=1}^n\frac{\left(x_i-x\right)^2}{\sigma_i^2} \\
+  \end{align*}
+  
+\end{frame}
+
+\begin{frame}
+  \frametitle{Оценки наибольшего правдоподобия}
+  \framesubtitle{Функции потерь}
+  Но что делать, если ошибки распределены по-другому?
+
+  Пусть, например, $\varepsilon_i$ подчинены распределению Коши.
+
+  \begin{align*}
+    \overline{x} & =\argmax_{x}\prod_{i=1}^n\frac{1}{\pi \left(x_i-x\right)^2} \\
+                 & =\argmin_{x}\sum_{i=1}^n \log\left(1+\left(x_i-x\right)^2\right) \\
+  \end{align*}
+  
+  Мы получили оценку, соответствующую функции потерь Коши.
+\end{frame}
+
+% \begin{frame}
```

Comment:
If it is not needed -- consider removing "dead" code from the file; otherwise -- consider completing this slide